com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl

```diff
@@ -15,9 +15,14 @@ SAMPLER(sampler_BaseMap);
 SAMPLER(sampler_ScreenSpaceOcclusionTexture);
 
 // Params
-float4 _BlurOffset;
 float4 _SSAOParams;
 float4 _SourceSize;
+float4 _ProjectionParams2;
+float4x4 _CameraViewProjections[2]; // This is different from UNITY_MATRIX_VP (platform-agnostic projection matrix is used). Handle both non-XR and XR modes.
+float4 _CameraViewTopLeftCorner[2];
+float4 _CameraViewXExtent[2];
+float4 _CameraViewYExtent[2];
+float4 _CameraViewZExtent[2];
 
```

Comment:
do you need 32 bit float precision for these uniforms and the ReconstructViewPos calculations? On mobile we are already doing too much fp32 bit math. Using half precision would be recommended (both for the uniforms + all the local variables)

Reply:
(this will be addressed as part of a future optimisation PR)

Reply:
I added a comment about potentially moving these uniforms to half.
Since the whole SSAO code does not use half at the moment, it would be a bigger task to optimize SSAO for half types.